src/steps/relay.py

```diff
@@ -34,6 +45,25 @@ def setup_main_relay_nodes(self, request):
         self.node2.start(relay="true", discv5_bootstrap_node=self.enr_uri)
         self.main_nodes.extend([self.node1, self.node2])
 
+    @pytest.fixture(scope="function")
+    def setup_main_rln_relay_nodes(self, request):
+        logger.debug(f"Running fixture setup: {inspect.currentframe().f_code.co_name}")
+        rln_creds = {
+            "eth_client_address": ETH_CLIENT_ADDRESS,
+            "go_waku_eth_client_private_key": GO_WAKU_ETH_TESTNET_KEY,
+            "nwaku_eth_client_private_key": NWAKU_ETH_TESTNET_KEY,
+            "keystore_password": KEYSTORE_PASSWORD,
+            "eth_contract_address": ETH_CONTRACT_ADDRESS,
+        }
+        self.node1 = WakuNode(NODE_1, f"node1_{request.cls.test_id}")
```

Comment:
I was thinking the registration itself could be a subject to test. And I was assuming we don't want to preserve registration or RLN state, so it would be fine to always start RLN test sequence with registration. What do you think ?  Checking presence of keystore.json would be sufficient check, I agree.

Reply:
Yes, we can keep a test just for registration check.
But other tests should still be able to run the registration (if it doesn't exist)
Basically the tests should be atomic and not depend on upstream tests

Reply:
I've added single registration test register_rln_single_node() as an @allure.step and calling it in the loop with test_register_rln()

Reply:
I've also added test to check for keystore presence accepting key_id 
https://github.com/waku-org/waku-interop-tests/pull/16/commits/864f63f5bb625b5db2c2ac6953971689a38bbcd6